File: app/src/main/java/net/daverix/ajvm/jvm/StringBuilderObject.java
package net.daverix.ajvm.jvm;


import java.io.IOException;

public class StringBuilderObject implements VirtualObject {
    private StringBuilder builder;

    public StringBuilderObject() {

    }

    @Override
    public String getName() {
        return "java/lang/StringBuilder";
    }

    @Override
    public void setFieldValue(String fieldName, Object value) {

    }

    @Override
    public Object getFieldValue(String fieldName) {
        return null;
    }

    @Override
    public Object invokeMethod(String name, String descriptor, Object... args) throws IOException {
        if("<init>".equals(name))  {
            builder = new StringBuilder();
            return null;
        } else if("append".equals(name)) {
            builder.append(args[0]);
            return this;
        } else if("toString".equals(name)) {
            return builder.toString();
        }

        throw new UnsupportedOperationException("method " + name +" not implemented in StringBuilderObject");
    }
}

File: core/src/commonMain/kotlin/net/daverix/ajvm/StringBuilderObject.kt
/*
    Java Virtual Machine for Android
    Copyright (C) 2017 David Laurell

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
 */
package net.daverix.ajvm

class StringBuilderObject : VirtualObject {
    private var builder: StringBuilder? = null

    override fun invokeMethod(name: String, descriptor: String, args: Array<Any?>): Any? {
        return when (name) {
            "<init>" -> {
                builder = StringBuilder()
                null
            }
            "append" -> {
                builder!!.append(args[0])
                this
            }
            "toString" -> builder!!.toString()
            else -> throw UnsupportedOperationException("method $name not implemented in StringBuilderObject")
        }
    }

    override fun getFieldValue(fieldName: String): Any? {
        error("no field with name $fieldName exist")
    }

    override fun setFieldValue(fieldName: String, value: Any?) {
        error("no field with name $fieldName exist")
    }
}
